roles/swift-ring/tasks/main.yml

```diff
@@ -13,3 +13,4 @@
   command: sudo -u swiftops /usr/local/bin/swifttool -i /home/swiftops/.ssh/id_rsa
            bootstrap --config /etc/swift/ring_definition.yml --outdir /etc/swift
   when: not should_boostrap|success
+  notify: restart swift services
```

Comment:
What if we order the roles so that the rings are created before the services start? Can that be done so that the rings are in place when the role that starts those services is executed?

As for dependency roles, the openstack-source role is a dependency role. That's how the source for the project gets installed. You can tell that role what extra things to put into the virtualenv. I would start with an unversioned reference to keystone-middleware and see if it functions.